src/test/java/ReceiptAnalysisTest.java

```diff
@@ -160,9 +162,31 @@ public void serveImageTextThrowsIfEmptyTextAnnotationsList()
     when(client.batchAnnotateImages(Mockito.<AnnotateImageRequest>anyList()))
         .thenReturn(batchResponse);
 
-    expectedException.expect(ReceiptAnalysisException.class);
-    expectedException.expectMessage(EMPTY_TEXT_ANNOTATIONS_LIST_WARNING);
+    ReceiptAnalysisException exception = Assertions.assertThrows(
+        ReceiptAnalysisException.class, () -> { ReceiptAnalysis.serveImageText(url); });
+
+    Assert.assertEquals(EMPTY_TEXT_ANNOTATIONS_LIST_WARNING, exception.getMessage());
+  }
+
+  @Test
+  public void serveImageTextThrowsIfRequestFails() throws IOException, ReceiptAnalysisException {
+    URL url = mock(URL.class);
+    InputStream inputStream = new ByteArrayInputStream(IMAGE_BYTES.toByteArray());
+    when(url.openStream()).thenReturn(inputStream);
+
+    ImageAnnotatorClient client = mock(ImageAnnotatorClient.class);
+    mockStatic(ImageAnnotatorClient.class);
+    when(ImageAnnotatorClient.create()).thenReturn(client);
```

Comment:
If these are the same in all your tests, you may want to put it in you setUp() method.